### PR TITLE
feat: migrate operator code base to support new operator-sdk version

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR updates the code base for compatibility with the latest operator-sdk version(v1.35.0)

Key Changes:
* Moved API files from `pkg/api` to `api` folder
* Relocated `main.go` from `cmd/manager/main.go` to `cmd/main.go`
* Shifted controllers from `pkg/controllers` to `controllers` folder
* Made minor updates to components, utils and tests to use correct import paths
* Updated scaffolding config to align with the latest operator-sdk version
* Revised Operator, bundle and e2e Dockerfiles
* Upgraded operator-sdk to v1.35.0 and controller-gen to v0.13.0
* Updated bundle manifests accordingly

### CODE REVIEW

File: bundle/manifests/kepler-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -28,14 +28,14 @@ metadata:
     capabilities: Seamless Upgrades
     categories: Monitoring
     containerImage: quay.io/sustainable_computing_io/kepler-operator:0.13.0
-    createdAt: "2024-05-22T07:06:13Z"
+    createdAt: "2024-06-24T08:00:23Z"
     description: 'Deploys and Manages Kepler on Kubernetes '
-    operators.operatorframework.io/builder: operator-sdk-v1.27.0
+    operators.operatorframework.io/builder: operator-sdk-v1.34.2
```
Comment (sthaha) [inline bundle/manifests/kepler-operator.clusterserviceversion.yaml:33]: How about we move to 1.35 ?

File: internal/controller/kepler_controller.go
Comment (vprashar2929) [inline internal/controller/kepler_controller.go:1]: AFAIU we can opt-out the internal folder as all the controller code should be in the project root dir https://sdk.operatorframework.io/docs/building-operators/golang/migration/#controllers 
Any thoughts?

File: bundle/tests/scorecard/config.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ stages:
   - entrypoint:
     - scorecard-test
     - basic-check-spec
-    image: quay.io/operator-framework/scorecard-test:v1.25.1
+    image: quay.io/operator-framework/scorecard-test:v1.34.2
```
Comment (sthaha) [inline bundle/tests/scorecard/config.yaml:11]: 1.35.0 ? :)

File: hack/bundle.sh [hunk 3]
```diff
@@ -58,7 +58,7 @@ main() {
 	info "Found old version: $old_bundle_version"
 
 	info "Building bundle version $VERSION"
-	run operator-sdk generate kustomize manifests --apis-dir=./pkg/api --verbose
+	run operator-sdk generate kustomize manifests --apis-dir=./api --verbose
```
Comment (sthaha) [inline hack/bundle.sh:61]: IIRC, api is the default dir. Could you please check?

File: internal/controller/kepler_controller.go
Comment (sthaha) [reply internal/controller/kepler_controller.go:1]: Lets not keep the controller internal. `internal` packages have special meaning - https://go.dev/doc/go1.4#internalpackages

File: Dockerfile [hunk 4]
```diff
@@ -12,28 +12,32 @@ COPY go.sum go.sum
...
-COPY must-gather/ must-gather/
-COPY cmd/ cmd/
+COPY cmd/main.go cmd/main.go
+COPY api/ api/
 COPY pkg/ pkg/
+COPY controllers/ controllers/
+
 
+# Copy must-gather scripts
+COPY must-gather/ must-gather/
```
Comment (sthaha) [inline Dockerfile:22]: lets preserve the order based on chance of modifying these files.

File: config/webhook/service.yaml [hunk 5]
```diff
@@ -19,3 +19,6 @@ spec:
     app.kubernetes.io/component: manager
     app.kubernetes.io/instance: controller-manager
     app.kubernetes.io/part-of: kepler-operator
+
+  # selector:
+  #   control-plane: controller-manager
```
Comment (sthaha) [inline config/webhook/service.yaml:24]: delete

File: PROJECT [hunk 6]
```diff
@@ -9,6 +13,7 @@ repo: github.com/sustainable.computing.io/kepler-operator
 resources:
 - api:
     crdVersion: v1
+    namespaced: true
```
Comment (sthaha) [inline PROJECT:16]: this shouldn't marked as namespaced. Our APIs are cluster-scoped.

File: config/rbac/service_account.yaml [hunk 7]
```diff
@@ -3,7 +3,7 @@ kind: ServiceAccount
 metadata:
   labels:
     app.kubernetes.io/name: serviceaccount
-    app.kubernetes.io/instance: controller-manager
+    app.kubernetes.io/instance: controller-manager-sa
```
Comment (sthaha) [inline config/rbac/service_account.yaml:6]: Have you tested upgrade path with this? Will there be 2 SA after we upgrade ?

Comment (vprashar2929) [reply config/rbac/service_account.yaml:6]: When the Operator is upgraded it doesn't change the name of the SA as the name is fixed to `controller-manager`

**Review by sthaha — APPROVED** [review]

**Pull request merged**